p2p/net/mock/mock_test.go

```diff
@@ -14,8 +14,8 @@ import (
 	protocol "github.com/ipfs/go-libp2p/p2p/protocol"
 	testutil "github.com/ipfs/go-libp2p/testutil"
 
-	context "gx/QmacZi9WygGK7Me8mH53pypyscHzU386aUZXpr28GZgUct/context"
 	detectrace "github.com/jbenet/go-detect-race"
+	context "gx/QmacZi9WygGK7Me8mH53pypyscHzU386aUZXpr28GZgUct/context"
```

Comment:
`gx/` doesn't encode protocol information re retrieval; @jbenet makes a good point here. However, the import path here doesn't **need** to encode protocol information. @whyrusleeping, I think we talked about something like a `.git` directory (so, `.gx`) that stored each package's local installation FS path (`gx/<hash>` here). And then the `package.json` would define that package's protocol path (e.g. `/ipns/<hash>`). My only qualm with adding another layer of nesting is that diminishes the user experience (try using command line tools in Java projects -- ha).

`gx/[ipfs|ipns]/`: encodes the protocol underneath the `gx/` prefix. Nice that it matches up with what's in `package.json`, but not necessary (as per above). I like the 1:1 matching of FS path and location here.

I don't really like `ipfs/hash/`, since drops the `gx` substring. Having it might be nice for identifying what this folder means when looking at a project using gx.